component/useProductList.js

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+
+const useProductList = () => {
+  const [products, setProducts] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [skip, setSkip] = useState(0);
+
+  const fetchProducts = async () => {
+    try {
+      const res = await fetch(`https://dummyjson.com/products?skip=${skip}`);
+      const data = await res.json();
+      setProducts((prevProducts) => [...prevProducts, ...data?.products]);
```

Comment:
For safety write like : setProducts(prevProducts => [...prevProducts, ...(data?.products || [])); because data?.products can be undefined or null and applying spread operator will result is error